crates/api/src/externals.rs

```diff
@@ -34,7 +34,7 @@ impl Extern {
     /// Returns the underlying `Func`, if this external is a function.
     ///
     /// Returns `None` if this is not a function.
-    pub fn func(&self) -> Option<&Func> {
+    pub fn func(self) -> Option<Func> {
```

Comment:
It looks like a lot of the diff is due to the change here, I'm curious how come this changed though?

Reply:
With `Instance::exports()` computing `Extern` values on demand, it now returns `Extern` instead of `&Extern`. In code like `let func = instance.get_export("foo").unwrap().func().unwrap()`, if `func()` took a `&self`, it'd be taking a reference to a temporary.

Reply:
Makes sense yeah, could we perhaps add alternative methods for these idioms though? Idiomatically this feels like the correct signature for this function (a by-value perhaps being `into_func(self)`). Perhaps we could add `Instance::functions()` and such which return iterators over just those values?

Reply:
Ok, renaming them to `into_func()`.

We could add `Instance::functions()` if that's a common use case, but users could also write `Instance.exports().filter_map(|export| export.into_func())` if that's what they want.